doc/lsst.meas.base/intro.rst

```diff
@@ -0,0 +1,94 @@
+See, for example, `here`_.
+These can be over-ridden by the user in the task configuration.
+
+When the driver runs, it will return an ``afw::table``.
+Each row in the table contains the results of running all of the plugins which have been enabled on a particular source.
+Each algorithm plugin can add columns to the tables to contain its results.
+These columns build upon the naming scheme described above.
+Thus, if the ``base_PsfFlux`` algorithm is run, you will see a bunch of fields in the table with names starting ``base_PsfFlux_`` — ``base_PsfFlux_instFlux``, ``base_PsfFlux_instFluxErr``, and so on.
+
+(Incidentally, note that this means that the schema of the output table depends on the particular plugins that were enabled at task configuration time. This makes it awkward to provide a single document that describes pipeline outputs — they vary with user configuration.)
+
+Algorithm fields and flags
+==========================
+
+Following from the above, when a new measurement driver (task) is created, it will:
+
+#. Create a new schema for its output;
+#. Hand that to each of the measurement algorithms (plugins) it's planning to
+   run, and ask them to add fields which will be used to store their outputs.
+
+In the limit of poor documentation, we can look at the code for the algorithm to understand what fields it's adding and what they mean.
+(Alternatively, we can inspect the schema of the output table.)
+
+Note that algorithms can be implemented in C++ or Python, so we need to be able to look at both.
+
+`Here is`_ an example in Python: the ``base_FPPosition`` plugin.
+When this plugin is created, it is given a ``schema`` argument, which will ultimately be the schema of the measurement driver outputs.
+It adds three things to the schema: fields to contain its outputs, and two flag fields::
+
+   schema.addField(name + "_flag", type="Flag", doc="Set to True for any fatal failure")
+   schema.addField(name + "_missingDetector_flag", type="Flag", doc="Set to True if detector object is missing")
+
+When you look at the output of any measurement driver that has executed ``base_FPPosition``, you should see those fields in the schema.
+And you can see that the flags are defined and (briefly) documented: if the algorithm fails for any reason, ``base_FPPosition_flag`` will be set to ``True``, and if it fails specifically because the detector is missing, ``base_FPPosition_missingDetector_flag`` will be set to ``True``.
+
+Several other plugins are defined in Python and follow the same pattern.  See, for example, ``base_Jacobian`` (`look here <https://github.com/lsst/meas_base/blob/35d32cdfa0559496d21b7de0310bd9161e120578/python/lsst/meas/base/plugins.py#L156>`_), ``base_PeakCentroid`` (`and here <https://github.com/lsst/meas_base/blob/35d32cdfa0559496d21b7de0310bd9161e120578/python/lsst/meas/base/plugins.py#L319>`_), etc.
+
+Of course, plugins which are complex or have to run fast are written in C++.
```

Comment:
Put this under a subhead: "Inspecting the schema code of C++ plugins"

Reply:
Don't agree with this one.

Reply:
(I disagreed with this and the next few suggestions until I understood what you were driving at, but in fact they're ok.)